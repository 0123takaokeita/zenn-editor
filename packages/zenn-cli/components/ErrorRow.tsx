import React from 'react';
import { ErrorMessage } from '@types';

type Props = { errorMessage: ErrorMessage };

export const ErrorRow: React.FC<Props> = ({ errorMessage }) => {
  let className = 'error-row';
  if (!errorMessage.isCritical) {
    className = className + ' ' + 'error-row--critical';
  }
  return (
    <div
      className={className}
      dangerouslySetInnerHTML={{ __html: errorMessage.message }}
    />
  );
};
