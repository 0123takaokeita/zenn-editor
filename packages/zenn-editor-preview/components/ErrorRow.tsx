import { ErrorMessage } from "@types";

const ErrorRow: React.FC<{ errorMessage: ErrorMessage }> = ({
  errorMessage,
}) => {
  let className = "error-row";
  if (!errorMessage.isCritical) {
    className = className + " " + "error-row--critical";
  }
  return (
    <div
      className={className}
      dangerouslySetInnerHTML={{ __html: errorMessage.message }}
    />
  );
};

export default ErrorRow;
