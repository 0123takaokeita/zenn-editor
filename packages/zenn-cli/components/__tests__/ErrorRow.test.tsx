import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { ErrorRow } from '@components/ErrorRow';
import { ErrorMessage } from '@types';

const renderer = createRenderer();
const dummyError: ErrorMessage = {
  isCritical: false,
  message: 'dummy Error!',
};

describe('<ErrorRow />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<ErrorRow errorMessage={dummyError} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
