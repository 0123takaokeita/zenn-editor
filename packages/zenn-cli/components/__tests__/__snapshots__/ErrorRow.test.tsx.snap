// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorRow /> should render and match the snapshot 1`] = `
<div
  className="error-row error-row--critical"
  dangerouslySetInnerHTML={
    Object {
      "__html": "dummy Error!",
    }
  }
/>
`;
